pkg/envinfo/storage.go

```diff
@@ -37,6 +37,24 @@ func (ei *EnvInfo) ValidateStorage(storage localConfigProvider.LocalStorage) err
 			return fmt.Errorf("storage with name %s already exists", storage.Name)
 		}
 	}
+
+	//check if specified container exists or not
+	containerExists := false
+	containers, err := ei.GetContainers()
+	if err != nil {
+		return err
+	}
+
+	for _, c := range containers {
+		if c.Name == storage.Container {
+			containerExists = true
+		}
+	}
+
+	if !containerExists {
+		return fmt.Errorf("specified container %s does not exist", storage.Container)
+	}
```

Comment:
When no container is provided, this will result in an error

```
odo create nodejs
Devfile Object Validation
 ✓  Checking devfile existence [78923ns]
 ✓  Creating a devfile component from registry: test [109018ns]
Validation
 ✓  Validating if devfile name is correct [124009ns]

Please use `odo push` command to create the component with source deployed
odo storage create 
 ✗  specified container  does not exist
```
We can instead run it only when a container is provided by the user.

Reply:
Done